src/components/Details/Details.tsx

```diff
@@ -5,7 +5,8 @@ import { NavLink } from 'react-router-dom';
 import previous from '../../images/previous.png';
 
 interface MarketProps {
-  markets: {
+  id: string,
+  selectedMarket?: {
     id: number;
```

Comment:
Wow! nice refactor here. It's very readable and direct to what we need passed to Details.